src/Sylius/Bundle/AdminBundle/Action/ResendShipmentConfirmationEmailAction.php

```diff
@@ -42,9 +42,11 @@ public function __construct(
 
     public function __invoke(Request $request): Response
     {
-        $shipmentId = $request->attributes->get('id');
+        $shipmentId = $request->attributes->get('id', '');
 
-        if (!$this->csrfTokenManager->isTokenValid(new CsrfToken($shipmentId, (string) $request->query->get('_csrf_token')))) {
+        if (!$this->csrfTokenManager->isTokenValid(
+            new CsrfToken($shipmentId, (string) $request->query->get('_csrf_token', '')),
+        )) {
```

Comment:
Why don't we just throw a bad request exception?

Reply:
Not really a 4XX since we're getting an attribute so it's a misconfiguration of the route itself. Also since it's targeting 1.12 I'd prefer minimizing behavioural changes.